File: array/2-ctci-array-left-rotation-v2.js
// https://www.hackerrank.com/challenges/ctci-array-left-rotation

"use strict";

const fs = require("fs");

// Complete the rotLeft function below.
function rotLeft(a, d) {
  console.log("TCL: rotLeft -> d", d);
  const newD = d % a.length;
  return a.slice(newD).concat(a.slice(0, newD));
}

function main() {
  const file = fs.readFileSync("./2-testcase.txt", "utf8");
  console.log("TCL: main -> file ", file.split("\n")[1].split(" "));
  const a = file.split("\n")[1].split(" ");
  const d = file.split("\n")[0].split(" ")[1];
  console.log(rotLeft(a, d));
}

main();

File: array/2-ctci-array-left-rotation-v2.py
#!/bin/python3
# https://www.hackerrank.com/challenges/ctci-array-left-rotation
# Complete the rotLeft function below.


def rotLeft(a, d, n):
    temp = a[0:d]
    for i in range(n-d):
        a[i] = a[i + d]

    for i in range(d):
        a[n-d+i] = temp[i]
    return a


if __name__ == '__main__':
    a = [1, 2]
    n = 2
    d = 3
    result = rotLeft(a, d, n)
    print(result)
